### ISSUE TITLE
surface.setSize() work bad in P3D and with FX2D

### ISSUE THREAD

**Issue by StanLepunK** [processing/processing#4763]
The refresh don't work in `P3D` after `surface.setSize()`, that's work fine in `default`, `P2D`, `JAVA2D` renderer, plus `surface.setSize()` don't work with `FX2D`

```
void setup() {
  size(300,300, P3D) ;
  // size(300,300, FX2D) ;
  surface.setSize(600,600) ;
}


void draw() {
  fill(125) ;
  rect(0,0, width,height) ;
  fill(0) ;
  ellipse(mouseX, mouseY, 50,50) ;
}
```
![capture d ecran 2016-12-05 a 15 54 10](https://cloud.githubusercontent.com/assets/3332269/20889599/d9c72088-bb03-11e6-9b58-934377e02bbe.png)

**Comment by StanLepunK** [comment]
Same behavior on windows, with Processing for win 64.
The bug is apparently in the 'stroke(,,,)' method. Because when 'noStroke()' is active there is no bug.

**Comment by StanLepunK** [comment]
@JakubValtar There is link possible with your pullrequest and this problem https://github.com/processing/processing/pull/4795

**Comment by JakubValtar** [comment]
@StanLepunK No, this is not related.

**Comment by StanLepunK** [comment]
@JakubValtar It's pity ; I wrote that because your pullrequest is about `FX` and `background`

**Comment by StanLepunK** [comment]
Hi @codeanticode you know if there is a reason for this bug ? You know if it's possible to solve it ? Because I use this method in my app, and I must a decision about that to change the direction to set my app if there is no posibility to solve it ? Thanks.

**Comment by StanLepunK** [comment]
Processing 3.2.4 / OS sierra / iMac 2012
this bug is still there.

**Comment by StanLepunK** [comment]
with Processing 3.3 on Mac OS sierra, the bug with `FX2D` leave, but still there with `P3D`

**Comment by StanLepunK** [comment]
Hello @benfry I don't know if your read this issue. I just poke you in case you don't, because I think is not a little bug, isn't it ?

**Comment by StanLepunK** [comment]
The bug is closed to be fix on Processing 3.3.1
```void setup() {
 size(300,300, P3D) ;
 surface.setSize(600,600) ; // bug when it's used here
}


void draw() {
  surface.setSize(600,600) ; // work nice when it's used here
  fill(255) ;
  rect(0,0, width,height) ;
  noFill() ;
  stroke(0) ;
  ellipse(mouseX, mouseY, 50,50) ;
}
```

**Comment by StanLepunK** [comment]
Closed this issue, that's work fine now. Processing 3.3.7
